crmsh/utils.py

```diff
@@ -2946,7 +2946,7 @@ def subprocess_run_auto_ssh_no_input(cmd, remote=None, user=None, **kwargs):
             user = 'root'
         return su_subprocess_run(
             local_user,
-            'ssh {} {}@{} sudo -H -u {} /bin/sh'.format(constants.SSH_OPTION, remote_user, remote, user),
+            '{} {} {}@{} sudo -H -u {} /bin/sh'.format(ssh_command(), constants.SSH_OPTION, remote_user, remote, user),
```

Comment:
When there is `core.hosts` in crm.conf, calling ssh here

Reply:
So you divided a simple problem unnecessary into 2 parts. If you just check `no_ssh` in `user_pair_for_ssh` and raise an exception when it is `yes`, both `_guess_user_for_ssh` and this line will be skipped.